src/toil/jobStores/aws/jobStore.py

```diff
@@ -301,13 +301,25 @@ def _writeToUrl(cls, readable, url):
         dstKey.set_contents_from_string(readable.read())
 
     @staticmethod
-    def _extractKeyInfoFromUrl(url):
+    def _extractKeyInfoFromUrl(url, existing=None):
         """
+        Extracts bucket and key from URL. The existing parameter determines if a
+        particular state of existence should be enforced. Note also that if existing
+        is not True and the key does not exist
+        :param existing: determines what the state
         :return: (bucket, key)
         """
         s3 = boto.connect_s3()
         bucket = s3.get_bucket(url.netloc)
-        key = bucket.new_key(url.path[1:])
+        key = bucket.get_key(url.path[1:])
+        if existing is True:
+            assert key is not None
+        elif existing is False:
+            assert key is None
```

Comment:
We can't use `assert` to check for user errors. It would be a user error to pass a URL to an absent key for import.

Add branch for `is None`. The `else` branch should just do assert False.